code/recipes/how-to-spawn-vim/src/main.rs

```diff
@@ -0,0 +1,98 @@
+// ANCHOR: all
+// ANCHOR: imports
+use ratatui::{
+    backend::CrosstermBackend,
+    crossterm::{
+        event::{self, Event, KeyCode, KeyEventKind},
+        terminal::{disable_raw_mode, enable_raw_mode, EnterAlternateScreen, LeaveAlternateScreen},
+        ExecutableCommand,
+    },
+    widgets::Paragraph,
+};
+use std::io::{stdout, Result};
+use std::process::Command;
+
+type Terminal = ratatui::Terminal<CrosstermBackend<std::io::Stdout>>;
+// ANCHOR_END: imports
+
+// ANCHOR: action_enum
+enum Action {
+    EditFile,
+    Quit,
+    None,
+}
+// ANCHOR_END: action_enum
+
+// ANCHOR: main
+fn main() -> Result<()> {
```

Comment:
Move the loop part into a run function. This helps make it easy to handle all the main loop failures as a single thing.

```rust
    let terminal = ratatui::init();
    let app_result = run(terminal);
    ratatui::restore();
    app_result
```